test/test_extended_models.py

```diff
@@ -73,10 +74,30 @@ def test_get_model_weights(name, weight):
     ],
 )
 def test_weights_copyable(copy_fn, name):
-    model_weights = models.get_model_weights(name)
-    for weights in list(model_weights):
-        copied_weights = copy_fn(weights)
-        assert copied_weights is weights
+    for weights in list(models.get_model_weights(name)):
+        # It is somewhat surprising that (deep-)copying is an identity operation here,
+        # but this is the default behavior of enums.
```

Comment:
Still LGTM but here and below could we just

- link to the official docs https://docs.python.org/3/howto/enum.html#enum-members-aka-instances
- clarify that this isn't a behaviour that we want to enforce and that `==` would be fine and even preferable, should we change it in the future